Sources/SwiftParser/Parameters.swift

```diff
@@ -160,11 +160,20 @@ extension Parser {
     let misplacedSpecifiers = parseMisplacedSpecifiers()
 
     let names = self.parseParameterNames()
-    let colon = self.consume(if: .colon)
-    let type: RawTypeSyntax?
-    if colon != nil {
-      type = self.parseType(misplacedSpecifiers: misplacedSpecifiers)
-    } else {
+    var colon = self.consume(if: .colon)
+    // try to parse the type regardless of the presence of the preceding colon
+    // because the parameter may be unnamed
+    // e.g. { [X] in } or { (:[X]) in }
```

Comment:
Are these relevant here? From the code of `parseParameterClause`, I feel like `parseClosureParameter` is not even called.
If it does, could you add test cases for these patterns?

Reply:
I may just trim the braces from the examples in the comments.